src/components/InteractionLayer/index.js

```diff
@@ -92,22 +92,21 @@ class InteractionLayer extends React.Component {
     touchY: null,
   };
 
+  componentDidMount() {
+    if (this.props.ruler.timestamp) {
+      this.setRulerPosition(this.props.ruler.timestamp);
+    }
+  }
+
   componentWillReceiveProps(nextProps) {
     const {
-      subDomainsByItemId: prevSubDomainsByItemId,
+      timeSubDomain: prevTimeSubDomain,
```

Comment:
Can you toss this in for me?
```suggestion
      // FIXME: Migrate this to `subDomainsByItemId`.
      timeSubDomain: prevTimeSubDomain,
```